src/public/media.ts

```diff
@@ -77,3 +81,233 @@ export function captureImage(callback: (error: SdkError, files: File[]) => void)
+    let actionName = generateGUID();
+    let helper: MediaHelper = {
+      mediaMimeType: this.mimeType,
+      assembleAttachment: [],
+    };
+    const params = [actionName, this.content];
+    GlobalVars.getMediaHandler = callback;
+    this.content && callback && sendMessageRequestToParent('getMedia', params);
+    function handleGetMediaRequest(response: string): void {
+      if (GlobalVars.getMediaHandler) {
+        let mediaResult: MediaResult = JSON.parse(response);
+        if (mediaResult.error) {
+          GlobalVars.getMediaHandler(mediaResult.error, null);
+        } else {
+          if (mediaResult.mediaChunk) {
+            if (mediaResult.mediaChunk.chunkSequence <= 0) {
+              let file = createFile(helper.assembleAttachment, helper.mediaMimeType);
+              GlobalVars.getMediaHandler(mediaResult.error, file);
+            } else {
+              let assemble: AssembleAttachment = decodeAttachment(mediaResult.mediaChunk, helper.mediaMimeType);
+              helper.assembleAttachment.push(assemble);
+            }
+          } else {
+            GlobalVars.getMediaHandler(
+              { errorCode: ErrorCode.INTERNAL_ERROR, message: 'data receieved is null' },
+              null,
+            );
+          }
+        }
+      }
+    }
+
+    GlobalVars.handlers['getMedia' + actionName] = handleGetMediaRequest;
+  }
+}
+
+/**
+ * Input parameter supplied to the select Media API
+ */
+export interface MediaInputs {
+  /**
+   * List of media types allowed to be selected
+   */
+  mediaType: MediaType;
+
+  /**
+   * max limit of media allowed to be selected in one go, current max limit is 10 set by office lens.
+   */
+  maxMediaCount: number;
+
+  /**
+   * Additional properties for customization of select media in mobile devices
+   */
+  imageProps?: ImageProps;
+}
+
+/**
+ *  All properties in ImageProps are optional and have default values in the platform
+ */
+export interface ImageProps {
+  /**
+   * Optional; Lets the developer specify the image source, more than one can be specified.
+   * Default value is both camera and gallery
+   */
+  sources?: Source[];
+
+  /**
+   * Optional; Specify in which mode the camera will be opened.
+   * Default value is Photo
+   */
+  startMode?: Mode;
+
+  /**
+   * Optional; indicate if inking on the selected Image is allowed or not
+   * Default value is true
+   */
+  ink?: boolean;
+
+  /**
+   * Optional; indicate if user is allowed to move between front and back camera
+   * Default value is true
+   */
+  cameraSwitcher?: boolean;
+
+  /**
+   * Optional; indicate if putting text stickers on the selected Image is allowed or not
+   * Default value is true
+   */
+  textSticker?: boolean;
+
+  /**
+   * Optional; indicate if image filtering mode is enabled on the selected image
+   * Default value is false
+   */
+  enableFilter?: boolean;
+}
+
+/**
+ * The modes in which camera can be launched in select Media API
+ */
+export const enum Mode {
```

Comment:
It will be called as Imageprops.Mode, for other media types we can have audioProps.audioModes

Reply:
Agreed. But since this is defined as an independent enum, how do you propose we add the corresponding ones for audio or video? What would those enum be called?

Reply:
For video, Mode will be used as it is. All image props are applicable for vidoes. For audio we can use AudioMode maybe? I am sorry I don't want to rename variables at this point unless we see it as urgent as the changes will need to be propagated in all the three platforms.